New Topic: C# with a few initial concepts and terms (#1017)
Codecademy/docs: content/c-sharp/concepts/operators/operators.md

---
Title: 'Operators'
Description: 'Operators are used to perform various operations on variables and values.'
Subjects:
  - 'Computer Science'
  - 'Code Foundations'
Tags:
  - 'Operators'
  - 'Arithmetic'
  - 'Comparison'
  - 'Logical'
CatalogContent:
  - 'learn-c-sharp'
  - 'paths/computer-science'
---

> **Operators** are used to perform various operations on variables and values. The standard arithmetic and assignment operators are the most familiar.

Reviewer: ```suggestion
**Operators** are used to perform various operations on variables and values.
```

Reviewer: Second sentence doesn't add too much value

## Syntax

> The following code snippet uses the assignment operator, `=`, to set `my_variable` to the value of `num1` and `num2` with an arithmetic operator acting on them. For example, if `operator` represented `*`, `my_variable` would be assigned a value of `num1 * num2`.

Reviewer 2: ```suggestion
The following code snippet uses the assignment operator, `=`, to set `my_variable` to the value of `num1` and `num2` with an arithmetic operator operating on them. For example, if `operator` represented `*`, `my_variable` would be assigned a value of `num1 * num2`.
```

> The following code snippet uses the assignment operator, `=`, to set `my_variable` to the value of `num1` and `num2` with an arithmetic operator operating on them. For example, if `operator` represented `*`, `my_variable` would be assigned a value of `num1 * num2`.

Reviewer: ```suggestion
The following code snippet uses the assignment operator, `=`, to set `myVariable` to the value of `num1` and `num2` with an arithmetic operator operating on them. For example, if `operator` represented `*`, `myVariable` would be assigned a value of `num1 * num2`.
```

```pseudo

> my_variable = num1 operator num2;

Reviewer: ```suggestion
myVariable = num1 operator num2;
```

```

Operators can be organized into the following groups:

- Arithmetic operators for performing traditional math evaluations.
- Assignment operators for assigning values to variables.
- Comparison operators for comparing two values.

> - Logical operators for combining Boolean values.

Reviewer: ```suggestion
- Logical operators for combining `Boolean` values.
```

> - [Bitwise operators](https://www.codecademy.com/resources/docs/general/bitwise-operator) for manipulating the bits of a number

Reviewer: ```suggestion
- [Bitwise operators](https://www.codecademy.com/resources/docs/general/bitwise-operator) for manipulating the bits of a number.
```

> - Bitwise operators for manipulating the bits of a number

Reviewer: Can we link https://www.codecademy.com/resources/docs/general/bitwise-operator here?

## Arithmetic Operators

C# has the following arithmetic operators:

> - Addition, `+`, which returns the sum of two numbers.

Reviewer 2: ```suggestion
Addition, `+`, returns the sum of two numbers.
```

- Subtraction, `-`, which returns the difference of two numbers.
- Multiplication, `*`, which returns the product of two numbers.
- Division, `/`, which returns the quotient of two numbers.
- Modulus, `%`, which returns the remainder of one number divided by another.

> The above operate on two values. C# also has two unary operators.

Reviewer: ```suggestion
The above operate on two values. C# also has two unary operators:
```

- Increment, `++`, which increments its single operand by one.
- Decrement, `--`, which decrements its single operand by one.

Unlike the other arithmetic operators, the increment and decrement operators change the value of their operand as well as return a value. They also return different results depending on if they precede or follow the operand. Preceding the operand returns the value of the operand before the operation. Following the operand returns the value of the operand after the operation.

```cs
int x = 5;
int y = x++; // x now equals 6, y equals 5
int z = --x; // x now equals 5, z equals 5
```

> ## Logical Operators

Reviewer 2: There should be a mention here that logical operators ALWAYS evaluate both operands and do not "short-circuit" them like a conditional operator.

- The `??=` operator assigns the value on the right to the variable on the left if the variable on the left is `null`.

## Comparison Operators

C# has the following comparison operators:

- Equal, `==`, for returning `true` if two values are equal.
- Not equal, `!=`, for returning `true` if two values are not equal.
- Less than, `<`, for returning `true` if the left value is less than the right value.
- Less than or equal to, `<=`, for returning `true` if the left value is less than or equal to the right value.
- Greater than, `>`, for returning `true` if the left value is greater than the right value.
- Greater than or equal to, `>=`, for returning `true` if the left value is greater than or equal to the right value.

> **Note:** for these comparison operators, if any operand is not a number (`Double.NaN` or `Single.NaN`) the result of the operation is `false`.

## Bitwise Operators

C# has the following operators that perform operations on the individual bits of a number.

> - Bitwise complement, `~`, reverses each bit in a number.

Reviewer 2: ```suggestion
Bitwise complement, `~`, inverts each bit in a number.
```
